### ISSUE TITLE
Suppress Deprecation Warning in `WebMvcAutoConfiguration__BeanDefinitions` Class

### ISSUE THREAD

**Issue by akefirad** [spring-projects/spring-boot#33202]
And probably other generated `BeanDefinitions` by AoT?
Example:
```java
    public static BeanDefinition getThemeResolverBeanDefinition() {
      Class<?> beanType = ThemeResolver.class;
      RootBeanDefinition beanDefinition = new RootBeanDefinition(beanType);
      beanDefinition.setInstanceSupplier(getThemeResolverInstanceSupplier());
      return beanDefinition;
    }
```

**Comment by snicoll** [comment]
What is the benefit of suppressing the deprecation automatically? I don't think we should be doing that for generated code.

**Comment by akefirad** [comment]
Well, the code is generated by the library, so there's no way for me to suppress it. That means we'd see the warning during compilation. I guess this would be an issue wherever the source code is used (e.g. code analysis tools maybe?)

**Comment by snicoll** [comment]
Yes, I understand that. I don't think your code analysis tool should be looking at generated code. In the absence of an actual problem, I am going to close this. I understand that the warning can be annoying but blindly (automatically) suppressing it is worse IMO.

**Comment by wilkinsona** [comment]
We discussed this today and we think that there would be some value in the generated code not producing any compilation warnings. The warnings aren't helpful as the user can't do anything to the generated code to fix them.

We could configure Boot to compile the code with `-Xlint:none` but the warnings will still be present when viewed in an IDE. To avoid them entirely, `@SuppressWarnings` would have to be generated at build time. This would have to be done in Spring Framework where the code generation is performed. We'll transfer this issue to Framework for their consideration.
